### PR TITLE
Don't set selected attributes when using delimiter.

### PR DESCRIPTION
**Pull request by pstibrany**
**What this PR does**: This PR modifies GCS object client to avoid asking for selected attributes when using `List` with delimiter. Problem is that using both delimiter and selected attributes returns empty result, because `cloud.google.com/go/storage` package doesn't put `prefixes` into list of returned fields.

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/chunk/gcp/gcs_object_client.go [hunk 1]
```diff
@@ -130,7 +137,7 @@ func (s *GCSObjectClient) List(ctx context.Context, prefix, delimiter string) ([
 		}
 
 		// When doing query with Delimiter, Prefix is the only field set for entries which represent synthetic "directory entries".
-		if attr.Name == "" {
+		if attr.Prefix != "" {
```
Comment (pracucci) [inline pkg/chunk/gcp/gcs_object_client.go:140]: May you confirm Prefix is not also set objects?

**Review by pracucci — APPROVED** [review]
LGTM with a question

File: pkg/chunk/gcp/gcs_object_client.go (hunk 1 above)
Comment (pstibrany) [reply pkg/chunk/gcp/gcs_object_client.go:140]: It is not an attribute returned in objects (“items”) in the API. Only Go exposes it via same struct.

See “Response” in https://cloud.google.com/storage/docs/json_api/v1/objects/list

Comment (pstibrany) [reply pkg/chunk/gcp/gcs_object_client.go:140]: And https://cloud.google.com/storage/docs/json_api/v1/objects for list of fields for objects.

**Pull request merged**